tests/plotting/test_prediction_trajectory.py

```diff
@@ -143,6 +143,39 @@ def test_cross_entropy_smoke(prediction_trajectory: PredictionTrajectory):
     assert ce_stat.stats.shape == (3, 10)
 
 
+def test_rank_smoke(prediction_trajectory: PredictionTrajectory):
+    traj = prediction_trajectory
+    rank_stat = traj.rank(show_ranks=True)
+
+    assert rank_stat.name == "Rank"
+    assert rank_stat.units == ""
+    assert rank_stat.trajectory_labels is None
+    assert rank_stat.sequence_labels is None
+    assert rank_stat.stats.shape == (3, 10)
+
+
+def test_rank_correctness():
+    # Test that the rank is correct.
+    log_probs = np.array(
+        [[[[0.1, 0.2, 0.3], [0.6, 0.5, 0.4]], [[0.85, 0.8, 0.9], [1.0, 1.1, 1.2]]]]
+    )
+    assert log_probs.shape == (1, 2, 2, 3)  # (batch, layer, seq, vocab)
+
+    traj = PredictionTrajectory(
+        log_probs=log_probs,
+        input_ids=np.ones((1, 2), dtype=np.int64),
+        targets=np.array([[0, 1]], dtype=np.int64),
+    )
+
+    rank_stat = traj.rank(show_ranks=True)
```

Comment:
Should you test that this errors when batch_size > 1 ?

Reply:
It should work with batch sizes greater than 1

Reply:
Then why is the batch size squeezed out in line 172?

Reply:
ah nevermind , it's averaged over in `prediction_trajectory.py:535`